src/domains/domain.h

```diff
@@ -172,6 +173,11 @@ class domaint
   /// Remove all symbolic path restrictions.
   virtual void remove_all_sympath_restrictions()=0;
 
+  virtual tpolyhedra_domaint *get_tpolyhedra_domain()
```

Comment:
❓ It's a bit questionable that this should really be in the base class.

Reply:
I agree, but want to allow arbitrary domain as the inner array domain and at the same time use some specific zones-array combinations. To do that, we need to be able to extract the template polyhedra domain from *any* domain.

I take it as the polyhedra domain is specific (and always present) and therefore having this in the base class is fine.